### ISSUE TITLE
Pretrained models

### ISSUE THREAD

**Issue by TylerCarberry** [junyanz/pytorch-CycleGAN-and-pix2pix#10]
Do you have any CycleGAN pretrained models available that use pytorch?

**Comment by junyanz** [comment]
We are training some models now, and we will upload them soon.

**Comment by TylerCarberry** [comment]
Any update on this?

**Comment by divyanshj16** [comment]
Any update??

**Comment by petergerten** [comment]
I would also be very interested? Anybody has a pretrained model?

**Comment by jingweiz** [comment]
Hey,
do you have any training curves? I tried your code and the results do not look too good ... would be great if you can provide the training curves for a comparison, thanks a lot!

**Comment by rednithin** [comment]
Where can I find a pre trained model for edges2shoes and edges2handbags?

**Comment by isaacmg** [comment]
Anyone have the monet2photo pre-trained model?

**Comment by hugozanini** [comment]
Did someone train the network for segmentation?

**Comment by junyanz** [comment]
We have a few pre-trained models available. More are coming. 
pix2pix: edges2shoes, sat2map, and facades_label2photo
CycleGAN: horse2zebra and zebra2horse

Please check out the [download_scripts](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/tree/master/pretrained_models),  [instructions(pix2pix)](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix#apply-a-pre-trained-model-pix2pix) and  [instructions(CyclceGAN)](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix#apply-a-pre-trained-model-cyclegan) for more details.

**Comment by linkerzhang** [comment]
Hi @junyanz  I'm trying to load the pretrained cyclegan (horse2zebra) model as below code template,

model = XXXModel()
model.load_state_dict(torch.load('latest_net_G.pth'))

dumb question please. which XXXModel should I use in your code base please? I'm assuming that it should be a child class of torch.nn.Module.

Thank you very much!

**Comment by junyanz** [comment]
we don't have these models available. You can improve the results with additional identity loss.

**Comment by wnls** [comment]
Hi, 
Since there's no pretrained model for sat2map, I'm using default params to train sat2map by running 
`python train.py --dataroot ./datasets/maps --name maps_cyclegan --model cycle_gan --no_dropout`
as suggested in README.

The generated images don't seem to be right. Here's a generated image at epoch 151. Am I missing something? Thank you!

(epoch151_fake_A.png, epoch151_fake_B.png)
![epoch151_fake_a](https://user-images.githubusercontent.com/3687000/40682134-d9eb8c58-633f-11e8-86d0-ca4897b7cadd.png)
![epoch151_fake_b](https://user-images.githubusercontent.com/3687000/40682156-e5a7d588-633f-11e8-9a78-d86be5b60527.png)

**Comment by chandakQ** [comment]
Could you please add a pre-trained model for apple2orange.

Regards,
Mukesh Chandak

**Comment by junyanz** [comment]
We will more models soon. @taesung89

**Comment by taesung89** [comment]
Added! Please use the script `pretrained_models/download_cyclegan_model.sh` to download the pretrained models.

**Comment by joeatton** [comment]
Do you have the pretrained pix2pix model available for colourisation

**Comment by junyanz** [comment]
We don't' have one for pix2pix. Maybe you can check out this [repo](https://github.com/richzhang/colorization-pytorch).

**Comment by willhunger** [comment]
@wnls I caught the same problem, how did you slove it?

**Comment by DmitryUnanyants** [comment]
> Hi @junyanz  I'm trying to load the pretrained cyclegan (horse2zebra) model as below code template,
> 
> model = XXXModel()
> model.load_state_dict(torch.load('latest_net_G.pth'))
> 
> dumb question please. which XXXModel should I use in your code base please? I'm assuming that it should be a child class of torch.nn.Module.
> 
> or could you save and share the whole model for reference please? :)
> 
> Thank you very much!
>   
>   Forget about it. Figured out from the test_model.py. :)

Can you write please how did you solve this problem?

**Comment by banzhuanworker** [comment]
Hi, @junyanz , when I click on the following URL https://people.eecs.berkeley.edu/~taesung_park/pytorch-CycleGAN-and-pix2pix/models/$FILE.pth, it shows "Resource not found". 
Could you please upload the pretrained models and related datasets again? Thanks!

**Comment by taesungp** [comment]
Hello, 

we've updated the [URLs to the pretrained models](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/scripts/download_cyclegan_model.sh). The models are now at http://efrosgans.eecs.berkeley.edu/cyclegan/pretrained_models/.

**Comment by sameerjvn-tech** [comment]
Hi @junyanz @taesungp, I would like to compare the performance of the CycleGANs with UNET generator v/s ResNet generator. I had a few questions:
1. Do you have training details of your pre-trained models? (How many epochs, loss curves, etc.)
2. Since the CycleGAN class is using the ResNet 9-blocks generator by default, I am assuming that the pre-trained models are for this generator. In such case, Do you have pre-trained models using the UNET generator? (256 or 128)
